backend/compact-connect/tests/smoke/military_affiliation_smoke_tests.py

```diff
@@ -0,0 +1,131 @@
+    )
+
+    assert (
+        post_api_response.status_code == 200
+    ), f'Failed to POST military affiliations record. Response: {post_api_response.json()}'
+
+    """
+    Response body should include S3 pre-sign url form in this format:
+    {
+    ...
+        'documentUploadFields': [
+            {
+                'fields': {
+                    'key': f'compact/{TEST_COMPACT}/provider/{provider_id}/document-type/military-affiliations'
+                    f'/{today}/1234#military_affiliation.pdf',
+                    'x-amz-algorithm': 'AWS4-HMAC-SHA256',
+                },
+                'url': 'https://provider-user-bucket.s3.amazonaws.com/',
+            }
+        ],
+        'fileNames': ['military_affiliation.pdf'],
+        'status': 'initializing',
+    }
+    """
+    post_api_response_json = post_api_response.json()
+    # Use the S3 pre-signed URL to upload a test file to the S3 bucket.
+    with open('../resources/test_files/military_affiliation.pdf', 'rb') as test_file:
+        files = {'file': (test_file.name, test_file)}
+        pre_signed_url = post_api_response_json['documentUploadFields'][0]['url']
+        pre_signed_fields = post_api_response_json['documentUploadFields'][0]['fields']
+        s3_upload_response = requests.post(pre_signed_url, files=files, data=pre_signed_fields, timeout=10)
+        assert (
+            s3_upload_response.status_code == 204
+        ), f'Failed to upload test file to S3. Response: {s3_upload_response.text}'
+
+    # Wait a couple of seconds for S3 event to trigger the lambda that processes the uploaded file
+    # and updates the military affiliation record status to 'active'.
+    time.sleep(2)
+
+    # Get the provider data from the GET '/v1/provider-users/me' endpoint.
+    get_provider_data_response = requests.get(API_URL + '/v1/provider-users/me', headers=headers, timeout=10)
+    assert (
+        get_provider_data_response.status_code == 200
+    ), f'Failed to GET provider data. Response: {get_provider_data_response.json()}'
+    # check the response for a top level 'militaryAffiliations' field and verify it has a record with an upload date of
+    # today and a status of 'active'
+    provider_data = get_provider_data_response.json()
+    military_affiliations = provider_data.get('militaryAffiliations')
+    assert military_affiliations, 'No military affiliations found in provider data'
+    today = datetime.now(tz=UTC).date().isoformat()
+    matching_military_affiliation = next(
+        (ma for ma in military_affiliations if datetime.fromisoformat(ma['dateOfUpload']).date().isoformat() == today),
+        None,
+    )
+    assert matching_military_affiliation, f'No military affiliation record found for today ({today})'
+    assert matching_military_affiliation['status'] == 'active', 'Military affiliation record is not active'
+
+    sys.stdout.write(f'Successfully added military affiliation record: {matching_military_affiliation}\n')
```

Comment:
Why this vs. just `print`?

Reply:
The Ruff linter flags us for use of `print` statements in code, so this is the workaround

Reply:
Ah strange. Can we use print, but suppress the linter on those lines?

I'm fine with keeping this as-is, too, but can you add a comment noting why you're not using print here? Can be brief, but it does stand out without some context.

Reply:
I've suppressed the linter and replaced the print statements to improve readability